lld/COFF/InputFiles.h

```diff
@@ -89,16 +89,16 @@ class InputFile {
   // An archive file name if this file is created from an archive.
   StringRef parentName;
 
-  // Returns .drectve section contents if exist.
-  StringRef getDirectives() { return directives; }
+  // Returns .drectve section(s) content if exist.
+  llvm::SmallVector<StringRef, 0>  getDrectves() { return directives; }
 
   COFFLinkerContext &ctx;
 
 protected:
   InputFile(COFFLinkerContext &c, Kind k, MemoryBufferRef m, bool lazy = false)
       : mb(m), ctx(c), fileKind(k), lazy(lazy) {}
 
-  StringRef directives;
+  llvm::SmallVector<StringRef, 0> directives;
```

Comment:
I admit this is nitpicky and I'm not entirely sure, but I think 1 would be a more ideal default allocation size here. Depending on what kind of build it is, it's not uncommon to have directives in almost every single object file.

Reply:
+1

Reply:
If I understand the type correctly, it allocates 64 bytes minimum on the stack -- however any items that equates to (once overhead is subtracted, of course). Having the code reflect the intent is useful, though.